File: acct-workspace-ws-spec/src/main/java/com/desolatetimelines/acct/workspace/ws/model/CurrencyTransfer.java
package com.desolatetimelines.acct.workspace.ws.model;

/**
 * Describes a transfer of currency between two accounts
 *
 * @param sourceAccountUUID The UUID of the account from which the amount is transferred
 * @param targetAccountUUID The UUID of the account to which the amount ins transferred
 * @param amount            The amount that is transferred, in the target account's currency
 */
public record CurrencyTransfer(
    String sourceAccountUUID,
    String targetAccountUUID,
    Double amount
) {

}

File: acct-gui-angular/src/app/model-acct/currency-transfer.ts
import { Account } from "./account";

/**
 * Describes a transfer of a currency amount between two accounts having the same currency
 */
export interface CurrencyTransfer {

    /**
     * The account from which the amount is transferred
     */
    sourceAccount : Account,

    /**
     * The account to which the amount ins transferred
     */
    targetAccount : Account,

    /**
     * The amount that is transferred
     */
    amount : number

}